### PR TITLE
Partitioning

### PR DESCRIPTION
**Pull request by erikwessman**

### CODE REVIEW

File: weave/Objects/CurveSpawner.tscn [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[gd_scene load_steps=2 format=3 uid="uid://bc6jqse1kuixs"]
```
Comment (theowiik) [inline weave/Objects/CurveSpawner.tscn:1]: Readded this scene 👹

File: weave/Scripts/Player.cs [hunk 2]
```diff
@@ -12,7 +12,7 @@ public partial class Player : CharacterBody2D
     [Signal]
     public delegate void PlayerShotBulletEventHandler(Node2D bullet, Vector2 globalPosition);
 
-    private const int MovementSpeed = 100;
+    private float MovementSpeed = 100f;
```
Comment (theowiik) [inline weave/Scripts/Player.cs:15]: _movementSpeed

File: weave/Scripts/Player.cs [hunk 3]
```diff
@@ -25,7 +25,7 @@ public partial class Player : CharacterBody2D
     public CurveSpawner CurveSpawner { get; private set; }
 
     [GetNode("CollisionShape2D")]
-    public CollisionShape2D CollisionShape2D { get; private set; }
+    public CollisionShape2D CollisionShape2D;
```
Comment (theowiik) [inline weave/Scripts/Player.cs:28]: Re-add property?

File: weave/Scripts/Player.cs [hunk 4]
```diff
@@ -64,4 +64,10 @@ private void Rotate(double delta)
         if (Controller.IsTurningLeft())
             RotationDegrees -= 120 * (float)delta;
     }
+
+    public float GetRadius()
+    {
+        var shape = (CircleShape2D)CollisionShape2D.Shape;
```
Comment (theowiik) [inline weave/Scripts/Player.cs:70]: We have this cached in CircleShape

File: weave/Scripts/Player.cs (hunk 2 above)
Comment (theowiik) [reply weave/Scripts/Player.cs:15]: I think you will get syntax highlighting for this if you have the sonarlint extension

File: weave/Scripts/Grid.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+{
+    private readonly int _nrRows;
+    private readonly int _nrCols;
+    private readonly int _width;
+    private readonly int _height;
+    private readonly float _cellWidth;
+    private readonly float _cellHeight;
+    private readonly List<List<Cell>> _cells = new();
+
+    public class Cell
```
Comment (theowiik) [inline weave/Scripts/Grid.cs:17]: Can make this a struct (will get performance I believe :eyes: )

File: weave/Objects/CurveSpawner.tscn (hunk 1 above)
Comment (theowiik) [reply weave/Objects/CurveSpawner.tscn:1]: (can be deleted safely)

**Review by theowiik — CHANGES_REQUESTED** [review]

File: weave/Scripts/Grid.cs (hunk 5 above)
Comment (erikwessman) [reply weave/Scripts/Grid.cs:17]: 👀

File: weave/Scripts/Main.cs [hunk 6]
```diff
@@ -90,17 +105,16 @@ private void SpawnPlayers()
...
             _players.Add(player);
         });
     }
 
-    private static bool IsIntersecting(Player player, IEnumerable<SegmentShape2D> segments)
+    private static bool IsPlayerIntersecting(Player player, ISet<SegmentShape2D> segments)
     {
         var position = player.CollisionShape2D.GlobalPosition;
-        var radius = player.CircleShape.Radius + Constants.LineWidth / 2f;
+        var radius = player.GetRadius() + (Constants.LineWidth / 2);
```
Comment (theowiik) [inline weave/Scripts/Main.cs:117]: Possible loss of fraction, should be divided by 2f

File: weave/Scripts/Player.cs [hunk 7]
```diff
@@ -9,6 +9,9 @@ namespace weave;
 [Instantiable(ObjectResources.PlayerScene)]
 public partial class Player : CharacterBody2D
 {
+    [Signal]
+    public delegate void PlayerShotBulletEventHandler(Node2D bullet, Vector2 globalPosition);
```
Comment (theowiik) [inline weave/Scripts/Player.cs:13]: Re-added

**Review by theowiik — CHANGES_REQUESTED** [review]

**Review by theowiik — APPROVED** [review]

**Pull request merged**